src/redux/actions/authActions.js

```diff
@@ -0,0 +1,143 @@
+import fetch from 'cross-fetch';
+
+export const SIGNUP = 'SIGN_UP';
+export const SIGNUP_REQUEST = 'REQUEST_SIGNUP';
+export const SIGNUP_FAILURE = 'SIGNUP_FAILURE';
+export const SIGNUP_SUCCESS = 'SIGNUP_SUCCESS';
+export const LOGOUT = 'LOGOUT';
+export const LOGIN = 'LOGIN';
+export const LOGIN_REQUEST = 'REQUEST_LOGIN';
+export const LOGIN_FAILURE = 'LOGIN_FAILURE';
+export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
+export const LOGOUT_REQUEST = 'LOGOUT_REQUEST';
+export const LOGOUT_FAILURE = 'LOGOUT_FAILURE';
+export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';
+
+function requestSignup(user) {
+  return {
+    type: SIGNUP_REQUEST,
+    user: user,
+    receivedAt: Date.now()
+  };
+}
+
+function signupSuccess(user, json) {
+  return {
+    type: SIGNUP_SUCCESS,
+    user: json,
+    receivedAt: Date.now()
+  };
+}
+
+function signupFailure() {
+  return {
+    type: SIGNUP_FAILURE,
+    user: null,
+    receivedAt: Date.now()
+  };
+}
+
+function requestLogin(user) {
+  return {
+    type: LOGIN_REQUEST,
+    user: user,
+    receivedAt: Date.now()
+  };
+}
+
+function loginSuccess(user, json) {
+  return {
+    type: LOGIN_SUCCESS,
+    user: json,
+    receivedAt: Date.now()
+  };
+}
+
+function loginFailure() {
+  return {
+    type: LOGIN_FAILURE,
+    user: null,
+    receivedAt: Date.now()
+  };
+}
+
+function requestLogout() {
+  return {
+    type: LOGOUT_REQUEST,
+    user: null,
+    receivedAt: Date.now()
+  };
+}
+
+function logoutSuccess() {
+  return {
+    type: LOGOUT_SUCCESS,
+    user: null,
+    receivedAt: Date.now()
+  };
+}
+
+function logoutFailure() {
+  return {
+    type: LOGOUT_FAILURE,
+    user: null,
+    receivedAt: Date.now()
+  };
+}
+
+export function signup(user) {
+  return dispatch => {
+    dispatch(requestSignup(user)); // Signup request process has begun...
+    return fetch('http://localhost:4201/signup', {
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      method: 'post',
+      body: JSON.stringify(user)
```

Comment:
Is a user a data type with a username and a password?

Reply:
Currently, he name wrongly implies what is being stored here. User is not a data type with username and a password. It's the response from our server upon a successful login/signup; i.e. {success: ..., message: ..., token: ...  }. I was planning to continue working on finding an ideal way to retrieve the user data upon signup/login and passing it through the store as a user data type, which would be handled here. Forgot to make a comment on this. Will continue on with this as planned now.

Reply:
Wait, I think there's something funky going on here. For a post request to sign up a user, we need the following information in the field:
```
{
	"email" : "{{newemail}}",
	"password" : "{{password}}",
	"f_name" : "{{newfname}}",
	"l_name" : "{{newlname}}"
}
```

So, "user" isn't the response from the server, it is the message sent to the server.